internal/backend/remote-state/consul/client.go

```diff
@@ -231,7 +231,11 @@ func (c *RemoteClient) Put(data []byte) error {
 		}
 		// transaction was rolled back
 		if !ok {
-			return fmt.Errorf("consul CAS failed with transaction errors: %v", resp.Errors)
+			message := ""
+			for _, respError := range resp.Errors {
+				message += respError.What
+			}
+			return fmt.Errorf("consul CAS failed with transaction errors: %s", message)
```

Comment:
I think that concatenating the error messages in this way will work for a single error. But if there are multiple errors returned it will squash them all together into the string, without delimiters. We could use `strings.Join` when fmt'ing the error string; or, since go-multierror is already included, we could use that to build the error:

```suggestion
			var resultErr error
			for _, respError := range resp.Errors {
				resultErr = append(resultErr, errors.New(respError.What))
			}
			return fmt.Errorf("consul CAS failed with transaction errors: %w", resultErr)
```